eclass/java-osgi.eclass

```diff
@@ -7,20 +7,29 @@
 # @AUTHOR:
 # Java maintainers <java@gentoo.org>
 # @BLURB: Java OSGi eclass
+# @SUPPORTED_EAPIS: 5 6 7
 # @DESCRIPTION:
 # This eclass provides functionality which is used by packages that need to be
 # OSGi compliant. This means that the generated jars will have special headers
 # in their manifests. Currently this is used only by Eclipse-3.3 - later we
 # could extend this so that Gentoo Java system would be fully OSGi compliant.
 
+case ${EAPI:-0} in
+	[567]) ;;
+	*) die "${ECLASS}: EAPI ${EAPI:-0} not supported" ;;
+esac
+
+if [[ -z ${_JAVA_OSGI_ECLASS} ]] ; then
+_JAVA_OSGI_ECLASS=1
+
 inherit java-utils-2
 
 # @ECLASS-VARIABLE: _OSGI_T
 # @INTERNAL
 # @DESCRIPTION:
 # We define _OSGI_T so that it does not contain a slash at the end.
-# According to Paludis guys, there is currently a proposal for EAPIs that
-# would require all variables to end with a slash.
+# (According to Paludis guys, there is currently a proposal for EAPIs that
+# would require all variables to end with a slash.)
```

Comment:
Chances are that this will never happen (we did the opposite).
So, either leave that comment alone, or remove it altogether.